### PR TITLE
enforce wheel size limits, README formatting in CI

### PR DESCRIPTION
**Pull request by jameslamb**
## Description

Contributes to https://github.com/rapidsai/build-planning/issues/110

Proposes adding 2 types of validation on wheels in CI, to ensure we continue to produce wheels that are suitable for PyPI.

* checks on wheel size (compressed),
  - *to be sure they're under PyPI limits*
  - *and to prompt discussion on PRs that significantly increase wheel sizes*
* checks on README formatting
  - *to ensure they'll render properly as the PyPI project homepages*
  - *e.g. like how https://github.com/scikit-learn/scikit-learn/blob/main/README.rst becomes https://pypi.org/project/scikit-learn/*

## Notes for Reviewers

This intentionally omits the libraries that are about to be moved to other repos.

* `cugraph-dgl` / `cugraph-pyg` (https://github.com/rapidsai/cugraph-gnn/pull/67)
* `nx-cugraph` (https://github.com/rapidsai/nx-cugraph/pull/19)

### CODE REVIEW

**Review by bdice — APPROVED** [review]
Approving with one suggestion.

File: python/cugraph/pyproject.toml [hunk 1]
```diff
@@ -88,3 +88,11 @@ requires = [
...
 dependencies-file = "../../dependencies.yaml"
 matrix-entry = "cuda_suffixed=true"
+
+[tool.pydistcheck]
+select = [
+    "distro-too-large-compressed",
+]
+
+# detect when package size grows significantly
+max_allowed_size_compressed = '1.5G'
```
[the remark this answers is not in the document]
Comment (jameslamb) [reply python/cugraph/pyproject.toml:98]: Yeah with the benefit of a night of sleep, I do think that's better and the little bit of extra complexity in the script is worth it. I'll do that here.

And put up a PR in `cuvs` doing that too... it had a similar huge threshold + huge difference across CUDA versions.

[the remark this answers is not in the document]
Comment (jameslamb) [reply python/cugraph/pyproject.toml]: Just put up a proposal for setting these thresholds, let me know what you think.

**Pull request merged**